server/src/main/java/org/opensearch/cluster/decommission/DecommissionController.java

```diff
@@ -74,6 +77,61 @@ public class DecommissionController {
         this.threadPool = threadPool;
     }
 
+    /**
+     * This method sends a transport call to retry decommission action, given that -
+     * 1. cluster_manager_node_timeout is not timed out
+     * 2. And executed when there was a cluster manager change
+     *
+     * @param decommissionRequest decommission request object
+     * @param startTime start time of previous request
+     * @param listener callback for the retry action
+     */
+    public void retryDecommissionAction(
+        DecommissionRequest decommissionRequest,
+        long startTime,
+        ActionListener<DecommissionResponse> listener
+    ) {
+        final long remainingTimeoutMS = decommissionRequest.timeout().millis() - (threadPool.relativeTimeInMillis() - startTime);
+        if (remainingTimeoutMS <= 0) {
+            String errorMsg = "cluster manager node timed out before retrying ["
+                + DecommissionAction.NAME
+                + "] for attribute ["
+                + decommissionRequest.getDecommissionAttribute()
+                + "] after cluster manager change";
+            logger.debug(errorMsg);
+            listener.onFailure(new OpenSearchTimeoutException(errorMsg));
+            return;
+        }
+        decommissionRequest.setRetryOnClusterManagerChange(true);
+        decommissionRequest.setTimeout(TimeValue.timeValueMillis(remainingTimeoutMS));
```

Comment:
you should be using Builder to construct the new request instead of updating it?

Reply:
Didn't construct a new request as technically this is the same request which is just getting executed. Let me know if you think we should create a new request

Reply:
Sure. Will create new request